plugins/modules/proxmox_backup_schedule.py

```diff
@@ -0,0 +1,229 @@
+    backup_action: 'update_vmid'
+
+- name: Removing backup setting based on VM name.
+  proxmox_backup_schedule:
+    api_user: 'myUser@pam'
+    api_password: '*******'
+    api_host: '192.168.20.20'
+    vm_name: 'VM Name'
+    backup_action: 'delete_vmid'
+
+- name: Removing backup setting based on VM ID.
+  proxmox_backup_schedule:
+    api_user: 'myUser@pam'
+    api_password: '*******'
+    api_host: '192.168.20.20'
+    vm_id: 'VM ID'
+    backup_action: 'delete_vmid'
+"""
+
+RETURN = """
+---
+backup_schedule:
+  description:
+    - If V(update_vmid), the backup_schedule will return True after adding the VM ID to the backup job.
+    - If V(delete_vmid), the backup_schedule will return a list of backup job IDs where the VM ID has existed after removing it.
+  returned: always, but can be empty
+  type: dict
+"""
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible_collections.community.general.plugins.module_utils.proxmox import (
+    proxmox_auth_argument_spec, ProxmoxAnsible, HAS_PROXMOXER, PROXMOXER_IMP_ERR)
+
+
+class ProxmoxSetVMBackupAnsible(ProxmoxAnsible):
+    # Getting backup sections
+    def get_cluster_bklist(self):
+        try:
+            backupSections = self.proxmox_api.cluster.backup.get()
+        except Exception as e:
+            self.module.fail_json(msg="Getting backup sections failed: %s" % e)
+        return backupSections
+
+    def get_cluster_specific_bkjobid(self, backup_id):
+        try:
+            specificBackupID = self.proxmox_api.cluster.backup.get(backup_id)
+        except Exception as e:
+            self.module.fail_json(msg="Getting specific backup ID failed: %s" % e)
+        return specificBackupID
+
+    def set_vmid_backup(self, backup_id, bk_id_vmids):
+        try:
+            self.proxmox_api.cluster.backup.put(backup_id, vmid=bk_id_vmids)
+        except Exception as e:
+            self.module.fail_json(msg="Setting vmid backup failed: %s" % e)
+        return
+
+    def get_vms_list(self):
+        try:
+            vms = self.proxmox_api.cluster.resources.get(type='vm')
+        except Exception as e:
+            self.module.fail_json(msg="Getting vms info from cluster failed: %s" % e)
+        return vms
+
+    # convert vm name to vm ID
+    def vmname_2_vmid(self, vmname):
+        vmInfo = self.get_vms_list()
+        vms = [vm for vm in vmInfo if vm['name'] == vmname]
+        return (vms[0]['vmid'])
+
+    # add vmid to backup job
+    def backup_update_vmid(self, vm_id, backup_id):
+        bk_id_info = self.get_cluster_specific_bkjobid(backup_id)
+
+        # If bk_id_info is a list, get the first item (assuming there's only one backup job returned)
+        if isinstance(bk_id_info, list):
+            bk_id_info = bk_id_info[0]  # Access the first item in the list
+        vms_id = bk_id_info['vmid'].split(',')
+        if vm_id in vms_id:
+            bksetInfo = []
+            return bksetInfo
```

Comment:
no need to define a variable for that, just go and:
```suggestion
            return []
```